elastic/beats
# [Heartbeat] Fix hint-based monitor gen

## What does this PR do?



Make hint generator more flexible by removing host/port matching requirements.
Closes https://github.com/elastic/beats/issues/17771.
Closes https://github.com/elastic/beats/issues/34214.

## Checklist




## How to test this PR locally


Deploy the following config and check that exactly one monitor is generated with an unmatched host/port:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name:  nginx-svc
  namespace: default
  labels:
    app:  nginx-svc
spec:
  selector:
    matchLabels:
      app: nginx-svc
  replicas: 1
  template:
    metadata:
      labels:
        app:  nginx-svc
    spec:
      containers:
      - name:  nginx-svc
        image: nginx:1.23.1-alpine
        imagePullPolicy: Always
        resources:
          requests:
            cpu: 300m
            memory: 200Mi
          limits:
            cpu: 300m
            memory: 200Mi
        ports:
        - containerPort: 3000
---
apiVersion: v1
kind: Service
metadata:
  name: nginx-svc
  namespace: default
  annotations:
    co.elastic.monitor/1.hosts: 127.0.0.1:30000
    co.elastic.monitor/1.schedule: '@every 5s'
    co.elastic.monitor/1.type: tcp
spec:
  selector:
    app: nginx-svc
  type: NodePort
  ports:
  - protocol: TCP
    port: 80
    nodePort: 30000
    name: main
```
